pkg/unidling/metrics/metrics.go

```diff
@@ -0,0 +1,76 @@
+package prometheus
+
+import (
+	"sync"
+
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	unidlingCountDesc = prometheus.NewDesc(
+		"openshift_unidling_count",
+		"The number of times an unidling occurred",
+		nil,
+		nil,
+	)
+	uc = unidlingCollector{}
+)
+
+type unidlingCollector struct {
+	isCreated  bool
+	createOnce sync.Once
+	createLock sync.Mutex
+
+	getUnidlingCountFunc func() uint64
+}
+
+func InitializeMetricsCollector(unidlingCountFunc func() uint64) {
```

Comment:
When Promethus handles the counter, no function is needed here.

```suggestion
func InitializeMetricsCollector() {
```